File: TelefQuito/02.CLIWEB/TELFQUITO_RESTFUL_JAVA_WEB/services/CompraService.js
const axios = require("axios");

class CompraService {
  constructor() {
    this.baseUrl = "http://localhost:8080/TELFQUITO_RESTFUL_JAVA/resources";
  }

  async comprarEfectivo(carrito, codCedula) {
    try {
      console.log("Comprando en efectivo", carrito, codCedula);
      const response = await axios.post(`${this.baseUrl}/wscompra/efectivo`, {
        carrito,
        codCedula,
      });
      return response.data.result;
    } catch (error) {
      console.error("Compra Efectivo Error:", error);
      throw error;
    }
  }

  async comprarCredito(carrito, codCedula, tarjetaCredito) {
    try {
      console.log("Comprando con crédito", carrito, codCedula, tarjetaCredito);
      const response = await axios.post(`${this.baseUrl}/wscompra/credito`, {
        carrito,
        codCedula,
        tarjetaCredito,
      });
      return response.data.result;
    } catch (error) {
      console.error("Compra Crédito Error:", error);
      throw error;
    }
  }
  async obtenerFactura(cedula) {
    try {
      const response = await axios.get(`${this.baseUrl}/wscompra/factura`, {
        params: { cedula },
      });
      return response.data;
    } catch (error) {
      console.error("Obtener Factura Error:", error);
      throw error;
    }
  }

  async consultarTablaAmortizacion(cedula) {
    try {
      const response = await axios.get(
        `${this.baseUrl}/wscompra/tablaAmortizacion`,
        {
          params: { cedula },
        }
      );
      return response.data;
    } catch (error) {
      console.error("Consultar Tabla Amortizacion Error:", error);
      throw error;
    }
  }
}

module.exports = new CompraService();

File: TelefQuito/02.CLIMOV/app/src/main/java/com/example/telfquito_restful_java/service/CompraService.java
package com.example.telfquito_restful_java.service;

import com.example.telfquito_restful_java.models.Carrito;
import com.example.telfquito_restful_java.models.FacturaModel;
import com.example.telfquito_restful_java.models.TablaModel;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface CompraService {
    @POST("wscompra/efectivo")
    Call<ResponseBody> comprarEfectivo(@Body Carrito carrito, @Query("cedula") String cedula);

    @POST("wscompra/credito")
    Call<ResponseBody> comprarCredito(@Body Carrito carrito, @Query("cedula") String cedula, @Query("plazoMeses") int plazoMeses);

    @GET("wscompra/factura")
    Call<List<FacturaModel>> obtenerFactura(@Query("cedula") String cedula);

    @GET("wscompra/tablaAmortizacion")
    Call<List<TablaModel>> consultarTablaAmortizacion(@Query("cedula") String cedula);
}
